src/LinearA.hs

```diff
@@ -615,7 +630,21 @@ transposeExpr scope expr cts = case expr of
         (e2Ctx, e2Scope, e2Map) = transposeExpr scope   e2 cts
         (e1Ctx, e1Scope, e1Map) = transposeExpr e2Scope e1 ((e2Map !) <$> vs')
     -- TODO: Relax
-    _  -> error "Binding non-linear values in transposition!"
+    _ -> error "Binding non-linear values in transposition!"
+  -- TODO(axch): Cover this with a test case and uncomment it
+  -- LetUnpackLin vs v body ->
+  --   ( bCtx . LetMixed [] [v'] (Tuple $ map Var vs') . vCtx
+  --   , vScope
+  --   , vMap <> (bMap `M.withoutKeys` (S.fromList vs))
+  --   )
+  --   where
+  --     (bCtx, bScope, bMap) = transposeExpr scope body cts
+  --     vs' = (bMap !) <$> vs
+  --     v':_ = freshVars bScope
+  --     (vCtx, vScope, vMap) = transposeExpr (scopeExt bScope [v']) (Var v) [v']
+  LetUnpack vs v body -> (LetUnpack vs v . bCtx, bScope, bMap)
```

Comment:
I thought that transposition was supposed to be defined on expressions that only do linear operations, so we shouldn't need this?

Reply:
Exactly.  If the `body` is linear, the whole `LetUnpack` is linear too, and may need to be transposed.  However, the transposition is trivial, because the variable being unpacked (`v`) is non-linear by assumption.